infer/environment.cc

```diff
@@ -771,6 +771,15 @@ core::TypePtr Environment::processBinding(core::Context ctx, cfg::Binding &bind,
                 histogramInc("dispatchCall.components", dispatched.components.size());
                 tp.type = dispatched.returnType;
                 for (auto &comp : dispatched.components) {
+                    if (comp.method.exists() && comp.method.data(ctx)->owner.exists() &&
```

Comment:
I believe that his prohibits valid use-cases.

```
class HasAbstracts
  sig {abstract.void}
  def self.foo
  end
end

class Impl < HasAbstracts

  sig {implementation.void}
  def self.foo
    1
  end
end

sig{params(r: T.class_of(HasAbstracts))
def x(r)
  r.foo
end


x(Impl)
```

Reply:
I took a few stabs at ways of implementing this as a static check, but I ended up just leaving it off, so this isn't checked statically at all now.